modules/F50_base_aggregator.sh

```diff
@@ -76,58 +80,100 @@ F50_base_aggregator() {
 
     if [[ "${#RESULTS[@]}" -gt 0 ]]; then
       print_output ""
-      print_output "[+] ""$FUNCTION"" - top 10 results:"
+      print_output "[+] ""$FUNCTION""/""$FUNCTION1"" - top 10 results:"
+      i=0 
       for LINE in "${RESULTS[@]}" ; do
-        SEARCH_TERM=$(echo "$LINE" | cut -d\  -f3)
+        SEARCH_TERM="$(echo "$LINE" | cut -d\  -f3)"
+        F_COUNTER="$(echo "$LINE" | cut -d\  -f1)"
+        SEARCH_TERM1="$(echo "${RESULTS1[$i]}" | cut -d\  -f3)"
+        F_COUNTER1="$(echo "${RESULTS1[$i]}" | cut -d\  -f1)"
         if [[ -f "$BASE_LINUX_FILES" ]]; then
+          # if we have the base linux config file we are checking it:
           if grep -q "^$SEARCH_TERM\$" "$BASE_LINUX_FILES" 2>/dev/null; then
-            #LINE=$(echo "$LINE" | sed -e 's/\ \+/\t/g')
-            print_output "$(indent "$(green "$LINE"" - common linux file: yes")")"
+            if grep -q "^$SEARCH_TERM1\$" "$BASE_LINUX_FILES" 2>/dev/null; then
+              printf "${GREEN}\t%-5.5s : %-15.15s : common linux file: yes${NC}\t||\t${GREEN}%-5.5s : %-15.15s : common linux file: yes${NC}\n" "$F_COUNTER" "$SEARCH_TERM" "$F_COUNTER1" "$SEARCH_TERM1" | tee -a "$LOG_FILE"
+            else
+              printf "${GREEN}\t%-5.5s : %-15.15s : common linux file: yes${NC}\t||\t${ORANGE}%-5.5s : %-15.15s : common linux file: no${NC}\n" "$F_COUNTER" "$SEARCH_TERM" "$F_COUNTER1" "$SEARCH_TERM1" | tee -a "$LOG_FILE"
+            fi  
           else
-            #LINE=$(echo "$LINE" | sed -e 's/\ \+/\t/g')
-            print_output "$(indent "$(orange "$LINE"" - common linux file: no")")"
-          fi
+            if grep -q "^$SEARCH_TERM1\$" "$BASE_LINUX_FILES" 2>/dev/null; then
+              printf "${ORANGE}\t%-5.5s : %-15.15s : common linux file: no${NC}\t\t||\t${GREEN}%-5.5s : %-15.15s : common linux file: yes${NC}\n" "$F_COUNTER" "$SEARCH_TERM" "$F_COUNTER1" "$SEARCH_TERM1" | tee -a "$LOG_FILE"
+            else
+              printf "${ORANGE}\t%-5.5s : %-15.15s : common linux file: no${NC}\t\t||\t${ORANGE}%-5.5s : %-15.15s : common linux file: no${NC}\n" "$F_COUNTER" "$SEARCH_TERM" "$F_COUNTER1" "$SEARCH_TERM1" | tee -a "$LOG_FILE"
+            fi  
+          fi  
         else
-          print_output "$(indent "$(orange "$LINE")")"
-        fi
+          printf "${ORANGE}\t%-5.5s : %-15.15s${NC}\t\t||\t${ORANGE}%-5.5s : %-15.15s${NC}\n" "$F_COUNTER" "$SEARCH_TERM" "$F_COUNTER1" "$SEARCH_TERM1" | tee -a "$LOG_FILE"
+        fi  
+        (( i++ ))
       done
-    fi
-  fi
+    fi  
+  fi 
 
   print_output ""
-  if [[ -f "$LOG_DIR"/"$CVE_AGGREGATOR_LOG" ]]; then
+  if [[ "$S30_VUL_COUNTER" -gt 0 || "$CVE_COUNTER" -gt 0 || "$EXPLOIT_COUNTER" -gt 0 ]]; then
     print_output "[*] Identified the following version details, vulnerabilities and exploits:"
     print_output "$(grep "\[+\] Found version details" "$LOG_DIR"/"$CVE_AGGREGATOR_LOG" 2>/dev/null)"
```

Comment:
`print_output` can't modify the string correctly (colors): 

![image](https://user-images.githubusercontent.com/17053360/105006721-5030ab00-5a37-11eb-85bf-8efc5e0ae7fd.png)

Reply:
I will clean this a bit